service/tcp.go

```diff
@@ -342,7 +341,8 @@ func (h *tcpHandler) handleConnection(ctx context.Context, outerConn transport.S
 	id, innerConn, authErr := h.authenticate(outerConn)
 	if authErr != nil {
 		// Drain to protect against probing attacks.
-		h.absorbProbe(outerConn, authErr.Status, proxyMetrics)
+		port := outerConn.LocalAddr().(*net.TCPAddr).Port
```

Comment:
This couples with TCP. What if the outer conn is not TCP, which may be the case when we nest stuff.
I've been actively trying to remove dependencies on TCP, and this re-adds one that I removed.

We should right logic on streams that may or may not be network connections as much as possible.

Port doesn't really work anymore though. I think we need some sort of listener id. It could be the listener address. So pass the listener id to NewTCPHandler, to replace the port number. absorbProbe should take a listener id.